module/zfs/qat_crypt.c

```diff
@@ -297,10 +295,9 @@ qat_crypt(qat_encrypt_dir_t dir, uint8_t *src_buf, uint8_t *dst_buf,
 	CpaStatus status = CPA_STATUS_SUCCESS;
 	Cpa16U i;
 	CpaInstanceHandle cy_inst_handle;
-	Cpa16U nr_bufs = (enc_len + PAGE_CACHE_SIZE - 1) / PAGE_CACHE_SIZE;
+	Cpa16U nr_bufs = (enc_len >> PAGE_SHIFT) + 1;
```

Comment:
nr_bufs also need consider the address is not page aligned, changes should be:
Cpa16U nr_bufs = (enc_len + ((long)src & ~PAGE_MASK)) >> PAGE_SHIFT) + 1;
Same change is also needed in qat_compress.c line 244 and 245.